pypiprivate/azure.py

```diff
@@ -0,0 +1,99 @@
+import logging
+import os
+
+from azure.storage.blob import BlobServiceClient, ContentSettings
+
+from pypiprivate.storage import Storage, guess_content_type
+
+logger = logging.getLogger(__name__)
+
+
+class AzureBlobClientMixin(object):
+
+    def __init__(self, connection_string, container):
+        super().__init__()
+        self._connection_string = connection_string
+        self._container = container
+        self._blob_service_client = None
+        self._container_client = None
+
+    @property
+    def container(self):
+        return self._container
+
+    @property
+    def blob_service_client(self):
+        if self._blob_service_client:
+            return self._blob_service_client
+        self._blob_service_client = BlobServiceClient.from_connection_string(self._connection_string)
+        return self._blob_service_client
+
+    @property
+    def container_client(self):
+        if self._container_client:
+            return self._container_client
+        self._container_client = self.get_container_client(self._container)
+        return self._container_client
+
+    def get_container_client(self, container_name):
+        return self.blob_service_client.get_container_client(container_name)
+
+
+class AzureBlobStorage(Storage, AzureBlobClientMixin):
+
+    def __init__(self, connection_string, container, prefix=None):
+        super().__init__(connection_string, container)
+        self.prefix = prefix
+
+    @classmethod
+    def from_config(cls, config):
+        storage_config = config.storage_config
+        env = config.env
+        container = storage_config['container']
+        connection_string = storage_config.get('connection_string') or env['AZURE_CONNECTION_STRING']
```

Comment:
`connection_string` is not mentioned in the example config. But let's not add it in text file as it's a secret credential. 

I think instead of using connection_string, can we use account name and SAS key separately. That way account name (which is non-secret) can be added to the pypi-private.cfg file and SAS key can be set as env var. Right now the account name is part of connection string which can't be tracked in the config file. What do you think about this? 

Also, can we prefix the env var with `PP_` for consistency with other storages?